test/core/util/json/json_object_loader_test.cc

```diff
@@ -804,6 +804,111 @@ TEST(JsonObjectLoader, MapFields) {
       << test_struct.status();
 }
 
+//
+// map<> with comparator tests
+//
+
+TEST(JsonObjectLoader, MapWithComparatorFields) {
+  struct StringCmp {
+    using is_transparent = void;
+    bool operator()(absl::string_view a, absl::string_view b) const {
+      return a < b;
+    }
+  };
+
+  struct TestStruct {
+    std::map<std::string, int32_t, StringCmp> value;
+    std::map<std::string, std::string, StringCmp> optional_value;
+    std::optional<std::map<std::string, bool, StringCmp>> std_optional_value;
+    std::unique_ptr<std::map<std::string, int32_t, StringCmp>> unique_ptr_value;
+
+    static const JsonLoaderInterface* JsonLoader(const JsonArgs&) {
+      static const auto* loader =
+          JsonObjectLoader<TestStruct>()
+              .Field("value", &TestStruct::value)
+              .OptionalField("optional_value", &TestStruct::optional_value)
+              .OptionalField("std_optional_value",
+                             &TestStruct::std_optional_value)
+              .OptionalField("unique_ptr_value", &TestStruct::unique_ptr_value)
+              .Finish();
+      return loader;
+    }
+  };
+  // Valid map.
+  auto test_struct = Parse<TestStruct>("{\"value\": {\"a\":1}}");
+  ASSERT_TRUE(test_struct.ok()) << test_struct.status();
+  EXPECT_THAT(test_struct->value,
+              ::testing::ElementsAre(::testing::Pair("a", 1)));
+  EXPECT_THAT(test_struct->optional_value, ::testing::ElementsAre());
+  EXPECT_FALSE(test_struct->std_optional_value.has_value());
+  EXPECT_EQ(test_struct->unique_ptr_value, nullptr);
+  // .find can be called with an absl::string_view because of the StringCmp
+  // comparator
+  EXPECT_EQ(test_struct->value.find(absl::string_view("a"))->second, 1);
```

Comment:
This doesn't seem useful to test here, since it's not really part of the JsonObjectLoader API -- this is just testing that `std::map<>` works with a standard comparator, which is not the purpose of this test.

Same thing for each field below.

Reply:
Those are the only checks I added that are different than the already-written map cases - is this entire test still valueable just to make sure the custom comparator doesn't break?

Reply:
Yes, the purpose of this test is to just make sure that it works to have a data member of type `std::map<>` with a custom comparator.  As long as this builds and the fields get populated properly with this type, then the test is verifying what it needs to.

Reply:
Sounds good, removed them